src/Task/Phan.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace GrumPHP\Task;
+
+use GrumPHP\Runner\TaskResult;
+use GrumPHP\Task\Context\ContextInterface;
+use GrumPHP\Task\Context\GitPreCommitContext;
+use GrumPHP\Task\Context\RunContext;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Phan task
+ */
+class Phan extends AbstractExternalTask
+{
+    /**
+     * @return string
+     */
+    public function getName()
+    {
+        return 'phan';
+    }
+
+    /**
+     * @return OptionsResolver
+     */
+    public function getConfigurableOptions()
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults(
+            [
+                'output_mode'  => 'text',
+                'output'       => null,
+                'config_file'  => '.phan/config.php',
+                'triggered_by' => ['php']
+            ]
+        );
+
+        $resolver->addAllowedTypes('output_mode', ['string']);
+        $resolver->addAllowedTypes('output', ['null', 'string']);
+        $resolver->addAllowedTypes('config_file', ['string']);
+        $resolver->addAllowedTypes('triggered_by', ['array']);
+
+        return $resolver;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function canRunInContext(ContextInterface $context)
+    {
+        return ($context instanceof GitPreCommitContext || $context instanceof RunContext);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function run(ContextInterface $context)
+    {
+        $config = $this->getConfiguration();
+        $files = $context->getFiles()->extensions($config['triggered_by']);
+        if (0 === count($files)) {
+            return TaskResult::createSkipped($this, $context);
+        }
+
+        $config = $this->getConfiguration();
+
+        $arguments = $this->processBuilder->createArgumentsForCommand('phan');
+
+        $arguments->addOptionalArgumentWithSeparatedValue('-k', $config['config_file']);
```

Comment:
Can you use the long argument names? That makes debugging a bit easier in the future :)